From dc72a687afb33aa282ffbf6d35bd89a35905dc51 Mon Sep 17 00:00:00 2001
From: Jarmo Isotalo <jamo@isotalo.fi>
Date: Sun, 25 Oct 2015 15:54:26 +0200
Subject: [PATCH] Don't require root when no network is used and no root
 permissions are needed, and fix issue running sandbox with network disabled

---
 web/webapp.rb | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/web/webapp.rb b/web/webapp.rb
index d2627d4..f333c90 100755
--- a/web/webapp.rb
+++ b/web/webapp.rb
@@ -22,7 +22,7 @@ class WebappProgram
   end
 
   def run
-    raise "This should be run as root." if Process.uid != 0
+    raise "This should be run as root." if network_enabled? && Process.uid != 0
 
     mkdir_p_for_tmc_user(Paths.lock_dir)
     mkdir_p_for_tmc_user(Paths.work_dir)
@@ -128,6 +128,8 @@ private
           end
         end
       end
+    else
+      block.call
     end
   end
 
-- 
1.9.1

